mdtraj/formats/pdb/pdbfile.py

```diff
@@ -201,7 +201,7 @@ def load_pdb(
         # cell is not absurdly high. Standard water density is ~55 M, which
         # yields a particle density ~100 atoms per cubic nm. It should be safe
         # to say that no particle density should exceed 10x that.
-        particle_density = traj.top.n_atoms / traj.unitcell_volumes[0]
+        particle_density = np.float64(traj.top.n_atoms) / np.float64(traj.unitcell_volumes[0])
```

Comment:
unitcell_lengths and unitcell angles are often inherited from the exact file you load in so the volume calculated is often in <float64, especially from a pdb.  I did try to fix it [upstream](https://github.com/mdtraj/mdtraj/pull/1928/files/c2a5d162b00a5f084b86abd63ed95d0f453f99a3#diff-1b104b5984a85d8b849914ff10c636e498d9b75091dc845b9595a740f6580503R877) during volume calculation, but it didn't seem to be enough to fix the test. 

Also since I apparently broke the tests on 1.26, I should probably begin rolling back on some of my changes just so it's at an easier place for other people to start.

Reply:
Yes, right. Looking at it, I do see that the `unitcell_lengths` and angles are coming from the format readers. Thanks for pointing that out! It looks like the (or formats or whatever I should be calling them) are making instances of trajectories (let me know if that isn't always the case, I saw Amber and LAMMPS do this). For the particular case of unitcell vectors and angles, it seems to me like we could cast to `np.float64` in the `Trajectory` object for things like unit cell by using a `property` and a setter to make sure unitcells are always `np.float64`.

In some cases, it looks like data types are already "hard-coded", e.g. LAMMPS explicitly sets its lengths to `np.float32`.

https://github.com/mdtraj/mdtraj/blob/036e0d5f514d7e0344a3d7d90248c0c1e3384673/mdtraj/formats/lammpstrj.py#L313

Just thinking through things, we could take the approach of property + setter in `Trajectory` to make sure `unitcell_lengths` etc are in `np.float64` for trajectory objects, or we could go into the code for each format and make sure they are using `np.float64`. I guess I would prefer the first option unless there is a need to change data types within the specific format objects (seems like we'd have to carefully review what is going on in each format reader).

I'd be curious to know why your solution of casting during the volume calculation didn't work :) Maybe catching it even further upstream (such as enforcing the type of unitcell_lengths would help, though we'd have to evaluate if there were any implications)

Reply:
It's been like a month but I recall that when I set all of the `unit_lengths` to float64 in the Trajectory object, when we do traj.save(), it will effectively write something different due to precision difference, causing some of the other conversion tests to fail.

Reply:
Ah, okay. Maybe I should actually try it out and see :) I'm wondering if it's going to be possible to maintain the exact same behavior without the same type promotion rules or if we'll eventually have to make some decisions that might change some numbers slightly.